### ISSUE TITLE
Error 404 when downloading the released binary

### ISSUE THREAD

**Issue by dhiller** [rajatjindal/krew-release-bot#51]
There was one case when the krew-release-bot [ran into a 404 error as he'd tried to download the released binary](https://github.com/kubevirt/kubectl-virt-plugin/runs/1590255901?check_suite_focus=true#step:4:8).

I suspect that this rather was a failure on the CDN of GitHub as the released binaries should already have been present. When I tried to download the file manually (by clicking the link in the browser window) it succeeded right away.

**Comment by rajatjindal** [comment]
oh. let  me see if some info in logs of  the  bot. 

thanks for  reporting it

**Comment by rajatjindal** [comment]
found logs for successful execution only. 

```bash

10:38:30 PM: time="2020-12-21T17:08:30Z" level=info msg="will operate in tempdir /tmp/krew-index-854983437"
10:38:30 PM: time="2020-12-21T17:08:30Z" level=info msg="Cloning https://github.com/kubernetes-sigs/krew-index.git"
10:38:30 PM: Enumerating objects: 4, done.
10:38:30 PM: Counting objects:  25% (1/4)
Counting objects:  50% (2/4)
Counting objects:  75% (3/4)
Counting objects: 100% (4/4)
Counting objects: 100% (4/4), done.
10:38:30 PM: Compressing objects:  25% (1/4)
Compressing objects:  50% (2/4)
Compressing objects:  75% (3/4)
Compressing objects: 100% (4/4)
Compressing objects: 100% (4/4), done.
10:38:30 PM: Total 3224 (delta 0), reused 0 (delta 0), pack-reused 3220
10:38:31 PM: time="2020-12-21T17:08:31Z" level=info msg="Adding remote local at https://github.com/krew-release-bot/krew-index.git"
10:38:31 PM: creating branch kubevirt-virt-kubectl-virt-plugin-v0.36.0time="2020-12-21T17:08:31Z" level=info msg="creating branch kubevirt-virt-kubectl-virt-plugin-v0.36.0"
10:38:31 PM: time="2020-12-21T17:08:31Z" level=info msg="validating ownership"
10:38:31 PM: time="2020-12-21T17:08:31Z" level=info msg="update plugin manifest with latest release info"
10:38:31 PM: creating branch kubevirt-virt-kubectl-virt-plugin-v0.36.0time="2020-12-21T17:08:31Z" level=info msg="pushing changes to branch kubevirt-virt-kubectl-virt-plugin-v0.36.0"
10:38:32 PM: creating branch kubevirt-virt-kubectl-virt-plugin-v0.36.0
10:38:34 PM: time="2020-12-21T17:08:34Z" level=info msg="submitting the pr"
10:38:34 PM: creating branch kubevirt-virt-kubectl-virt-plugin-v0.36.0creating branch kubevirt-virt-kubectl-virt-plugin-v0.36.0time="2020-12-21T17:08:34Z" level=info msg="creating pr with title \"\\\"release new version v0.36.0 of virt\\\"\", \nhead \"\\\"krew-release-bot:kubevirt-virt-kubectl-virt-plugin-v0.36.0\\\"\", \nbase \"master\", \nbody \"\\\"hey krew-index team,\\n\\nI am [krew-release-bot](https://github.com/rajatjindal/krew-release-bot), and I would like to open this PR to publish version `v0.36.0` of `virt` on behalf of @dhiller.\\n\\nThanks,\\n@krew-release-bot\\\"\""
10:38:36 PM: time="2020-12-21T17:08:36Z" level=info msg="pr \"https://github.com/kubernetes-sigs/krew-index/pull/963\" opened for releasing new version"
10:38:36 PM: Duration: 5534.08 ms	Memory Usage: 72 MB	Init Duration: 148.72 ms	

```

**Comment by rajatjindal** [comment]
i will close this for now as it does look like a CDN problem. If it happens again and more frequently, then we will explore the option of retrying etc.

**Comment by dhiller** [comment]
/reopen

Had this again twice now:
* https://github.com/kubevirt/kubectl-virt-plugin/runs/1859240444?check_suite_focus=true#step:4:8
* https://github.com/kubevirt/kubectl-virt-plugin/runs/1780812104?check_suite_focus=true#step:4:8

**Comment by dhiller** [comment]
This has prevented the release tonight and a version update that should have happened. Can you please take a look? Thanks!

**Comment by dhiller** [comment]
Fortunately I could restart the GitHub actions by hand, so no pressure :)

**Comment by rajatjindal** [comment]
Hi @dhiller 

I will add some retry mechanism to handle this kind of transient failure.

**Comment by rajatjindal** [comment]
Hi @dhiller 

sorry for the long wait, i've released v0.0.39 with this fix. now it will retry 4 times with backoff if the released tar file is not found.

in anycase if this happens again, please let me know

**Comment by dhiller** [comment]
Thanks for taking the time fixing this! :bow:
